src/stores/standalone/notifications.ts

```diff
@@ -0,0 +1,128 @@
+//  Copyright (C) 2023 Nethesis S.r.l.
+//  SPDX-License-Identifier: GPL-3.0-or-later
+
+import { ref } from 'vue'
+import { defineStore } from 'pinia'
+import { uid } from 'uid/single'
+import { getAxiosErrorMessage, type NeNotification } from '@nethserver/vue-tailwind-lib'
+import { useI18n } from 'vue-i18n'
+import { useLoginStore } from './standaloneLogin'
+
+const NOTIFICATIONS_LIMIT = 30
+const DEFAULT_NOTIFICATION_TIMEOUT = 5000
+const ERROR_NOTIFICATION_TIMEOUT = 10000
+
+export const useNotificationsStore = defineStore('notifications', () => {
+  const { t } = useI18n()
+  const loginStore = useLoginStore()
+  const notifications = ref<NeNotification[]>([])
+  const isAxiosErrorModalOpen = ref(false)
+  const axiosErrorNotificationToShow = ref<NeNotification>()
+  const isNotificationDrawerOpen = ref(false)
+
+  const addNotification = (notification: NeNotification) => {
+    notifications.value.unshift(notification)
+    setNotificationShown(notification.id, true)
+
+    // limit total number of notifications
+    notifications.value = notifications.value.slice(0, NOTIFICATIONS_LIMIT)
+  }
+
+  const getNotificationTimeout = (notification: NeNotification) => {
+    return ['error', 'warning'].includes(notification.kind)
+      ? ERROR_NOTIFICATION_TIMEOUT
+      : DEFAULT_NOTIFICATION_TIMEOUT
+  }
+
+  const createNotificationFromAxiosError = (axiosError: any) => {
+    const notificationTitle = axiosError.config.url.includes('/ubus/call')
+      ? t('notifications.ubus_call_failed')
+      : t('notifications.api_call_failed')
+
+    const notification: NeNotification = {
+      id: uid(),
```

Comment:
May I suggest instead of using UIDs (and an additional library), just using an incremental ID?
Example: https://github.com/NethServer/ns8-user-manager/blob/main/src/stores/useNotificationEngine.ts

Reply:
I'd prefer to stick with random IDs, notification will be soon maintained inside local storage so we cannot reset the incremental ID anytime the page is refreshed